spec/ics-027-interchain-account/README.md

```diff
@@ -0,0 +1,296 @@
+
+### Port & channel setup  
+
+The `setup` function must be called exactly once when the module is created (perhaps when the blockchain itself is initialized) to bind to the appropriate port and create an escrow address (owned by the module).  
+
+```typescript  
+function setup() {  
+  relayerModule.bindPort("interchain-account", ModuleCallbacks{  
+    onChanOpenInit,  
+    onChanOpenTry,  
+    onChanOpenAck,  
+    onChanOpenConfirm,  
+    onChanCloseInit,  
+    onChanCloseConfirm,  
+    onSendPacket,  
+    onRecvPacket,  
+    onTimeoutPacket,  
+    onAcknowledgePacket,  
+    onTimeoutPacketClose  
+  })  
+}  
+```
+
+Once the `setup` function has been called, channels can be created through the IBC relayer module between instances of the interchain account module on separate chains.  
+
+An administrator (with the permissions to create connections & channels on the host state machine) is responsible for setting up connections to other state machines & creating channels to other instances of this module (or another module supporting this interface) on other chains. This specification defines packet handling semantics only, and defines them in such a fashion that the module itself doesn't need to worry about what connections or channels might or might not exist at any point in time.
+
+### Routing module callbacks  
+
+### Channel lifecycle management  
+
+Both machines `A` and `B` accept new channels from any module on another machine, if and only if:  
+
+- The other module is bound to the "interchain account" port.  
+- The channel being created is ordered.  
+- The version string is empty.  
+
+```typescript  
+function onChanOpenInit(  
+  order: ChannelOrder,  
+  connectionHops: [Identifier],  
+  portIdentifier: Identifier,  
+  channelIdentifier: Identifier,  
+  counterpartyPortIdentifier: Identifier,  
+  counterpartyChannelIdentifier: Identifier,  
+  version: string) {  
+  // only ordered channels allowed  
+  abortTransactionUnless(order === ORDERED)  
+  // only allow channels to "interchain-account" port on counterparty chain  
+  abortTransactionUnless(counterpartyPortIdentifier === "interchain-account")  
+  // version not used at present  
+  abortTransactionUnless(version === "")  
+}  
+```
+
+```typescript  
+function onChanOpenTry(  
+  order: ChannelOrder,  
+  connectionHops: [Identifier],  
+  portIdentifier: Identifier,  
+  channelIdentifier: Identifier,  
+  counterpartyPortIdentifier: Identifier,  
+  counterpartyChannelIdentifier: Identifier,  
+  version: string,  
+  counterpartyVersion: string) {  
+  // only ordered channels allowed  
+  abortTransactionUnless(order === ORDERED)  
+  // version not used at present  
+  abortTransactionUnless(version === "")  
+  abortTransactionUnless(counterpartyVersion === "")  
```

Comment:
For future compatibility, I think it is better to remove version checking logic if we are not actually using version
```suggestion

```